venues/aclweb.org/ACL/migrate_submissions_ARR_ACL.py

```diff
@@ -0,0 +1,344 @@
+# Create an empty list to store information about each paper for the PCs to view 
+submission_output_dict = {}
+for note in commitment_notes:
+    arr_submission_forum = note.content['paper_link'].split('=')[1]
+    submission_output_dict[note.forum] = {'acl_blind_submission_forum': None, 'arr_submission_forum': None, 'num_reviews':None, 'num_metareviews':None, 'is_latest_version':None, 'was_migrated':False}
+
+    later_duplicates = list(openreview.tools.iterget_notes(client, invitation = 'aclweb.org/ACL/ARR/2021/.*',content = {'previous_URL':f'https://openreview.net/forum?id={arr_submission_forum}'}))  
+    if later_duplicates:
+        submission_output_dict[note.forum]['is_latest_version'] = False
+    else:
+        submission_output_dict[note.forum]['is_latest_version'] = True
+    # Get the arr submission forum. Then check if that is in the submission dictionary. If it is, check for blind. If not, run post_acl
+    
+    if arr_submission_forum not in acl_submission_dict:
+        post_acl_submission(arr_submission_forum, note, submission_output_dict)
+        #print(f"{submission_output_dict['acl_commitment_forum']},{submission_output_dict['acl_blind_submission_forum']},{submission_output_dict['arr_submission_forum']},{submission_output_dict['num_reviews']},{submission_output_dict['num_metareviews']}, {submission_output_dict['is_latest_version']}, {submission_output_dict['was_migrated']}")
+        
+    # Check if the acl_submission.id is in blind_submissions (it is in acl_submissions)
+    elif acl_submission_dict[arr_submission_forum].id not in blind_submissions:
+        submission_output_dict[note.forum]['arr_submission_forum'] = arr_submission_forum
+        submission_output_dict[note.forum]['was_migrated'] = True
+        post_blind_submission(acl_submission_dict[arr_submission_forum].id, acl_submission_dict[arr_submission_forum], client.get_note(arr_submission_forum), submission_output_dict, note)
+        #print(f"{submission_output_dict['acl_commitment_forum']},{submission_output_dict['acl_blind_submission_forum']},{submission_output_dict['arr_submission_forum']},{submission_output_dict['num_reviews']},{submission_output_dict['num_metareviews']}, {submission_output_dict['is_latest_version']}, {submission_output_dict['was_migrated']}")
+
+    # Check if the blind submission has reviews (it is in blind submissions)
+    else: 
+        submission_output_dict[note.forum]['arr_submission_forum'] = arr_submission_forum
+        submission_output_dict[note.forum]['acl_blind_submission_forum'] = (acl_submission_dict[note.content['paper_link'].split('=')[1]]).forum
+        submission_output_dict[note.forum]['was_migrated'] = True
+        post_reviews(blind_submissions[acl_submission_dict[arr_submission_forum].id].forum, blind_submissions[acl_submission_dict[arr_submission_forum].id], client.get_note(arr_submission_forum), submission_output_dict, note)
```

Comment:
I'm not sure why this is an else of the previous conditions, shouldn't we need to call this for every commitment note.

Reply:
Each function calls the next, so if an acl submission was not posted for that commitment not it would go through post_submission, post_blind, post_reviews, post_metareviews

Reply:
I would prefer to keep all the functions independent and don't repeat the function calls:

```
acl_submission = post_acl_submission()
blind_acl_submission = post_blind_submission(acl_submission)
reviews = post_reviews(blind_acl_submission)
```

Each function can check of the note was already created or not like what you are doing in post_reviews. 

We can discuss this tomorrow.